hazardous/_survival_boost.py

```diff
@@ -572,6 +572,131 @@ def predict_survival_function(self, X, times=None):
         """
         return self.predict_cumulative_incidence(X, times=times)[:, 0, :]
 
+    def predict_incidence_after_s(self, X, censored_times, times=None):
+        r"""Estimate the conditional incidence functions for each event type
+        knowing that the individual has been censored at a given time :math:`s` i.e.
+        has survived until this time :math:`s`.
+
+        For each incidence function, :math:`\forall s > t`, we return:
+
+        .. math::
+
+            F_k(t| x_i, T > s) = \mathbb{P}(T \leq t, \Delta = k| X=x_i, T > s)
+            = \frac{\mathbb{P}(s < T \leq t, \Delta = k| X=x_i)}
+            {\mathbb{P}(T > s| X=x_i)}
+            = \frac{F_k(t| x_i) - F_k(s| x_i)}{S(s| x_i)}
+
+        And, :math:`\forall s > t`:
+
+        .. math::
+
+            S(t| x_i, T > s) = \mathbb{P}(T \geq t| X=x_i, T > s)
+            = \frac{\mathbb{P}(T \geq t| X=x_i)}
+            {\mathbb{P}(T > s| X=x_i)}
+            = \frac{S(t| x_i)}{S(s| x_i)}
+
+        where :math:`F_k(t| x_i)` is the estimated cumulative incidence function for
+        event type :math:`k` at time :math:`t` and :math:`S(s| x_i)` is the estimated
+        survival function at time :math:`s`.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The feature vectors for each observation for which to estimate the
+            incidence functions knowing the censoring time.
+
+        censored_times : array-like of shape (n_samples,)
+            The time of censoring for each observation.
+
+        times : array-like, default=None
+            The time horizons at which to estimate the probabilities. If ``None``,
+            it uses the grid generated during ``fit`` based on the parameter
+            ``n_time_grid_steps``.
+
+        Returns
+        -------
+        incidence_functions_knowing_censoring_time : ndarray of shape
+            (n_samples, n_events + 1, n_times) The estimated probabilities of the
+            different time horizons. For the times that are lower than the observed
+            event times, the survival function is 1, and the incidence functions will
+            be 0.
+        """
+        if times is None:
+            times = self.time_grid_
+        incidence_functions_knowing_censoring_time = self.predict_cumulative_incidence(
+            X, times=times
+        )
+
+        X_with_time = np.hstack([censored_times[:, None], X])
```

Comment:
This implementation is very nice. Before that, we must check that `censored_times` is iterable and turn it into an np.array. Then, we need to check that its shape matches X's, and raise informative error messages.
```python
from types import Iterable

if isinstance(censored_times, (str, bytes)) or not isinstance(censored_times, Iterable):
    raise ValueError(...)

censored_times = np.asarray(censored_times)

if censored_times.ndim != 1 or censored_times.shape[0] != X.shape[0]:
    raise ValueError(...)

censored_times = censored_times[:, None]
```
and then readapt the rest of the code accordingly